src/platform/utilities/eBenefitsUrl.js

```diff
@@ -0,0 +1,58 @@
+import {
+  ssoe,
+  ssoeEbenefitsLinks,
+} from 'platform/user/authentication/selectors';
+import { hasSessionSSO } from 'platform/user/profile/utilities';
+import environment from 'platform/utilities/environment';
+import { eauthEnvironmentPrefixes } from 'platform/utilities/sso/constants';
+
+const eauthPrefix = eauthEnvironmentPrefixes[environment.BUILDTYPE];
+// mapping of the www.ebenefits.va.gov/<path> path to the same page behind
+// the eauth.va.gov/<path> proxy. NOTE: if the path mapping is not defined
+// it is assumed that the paths between the two domains are the same
+const eauthPathMap = {
+  'ebenefits-portal/ebenefits.portal': 'ebenefits/homepage',
+  'ebenefits/about/feature?feature=disability-compensation':
+    'ebenefits/vdc?target=%2Fwssweb%2FVDC526%2Fcompensation.do',
+  'ebenefits/about/feature?feature=vocational-rehabilitation-and-employment':
+    'ebenefits/vre',
+  'ebenefits/about/feature?feature=direct-deposit-and-contact-information':
+    'ebenefits/manage/contact',
+  'ebenefits/about/feature?feature=payment-history': 'ebenefits/payments',
+  'ebenefits/about/feature?feature=dependent-compensation':
+    'ebenefits/vdc?target=%2Fwssweb%2Fwss-686-webparts%2Fdependent.do',
+  'ebenefits/about/feature?feature=cert-of-eligibility-home-loan':
+    'ebenefits/coe',
+  // NOTE: Future use; these links only exist in the content repo, so they
+  // can't be dynamically created.  However if those pages get moved over
+  // they we will have them mapped
+  'ebenefits/about/feature?feature=request-vso-representative':
+    'ebenefits/vdc?target=%2Fwssweb%2FVDC2122%2Frepresentative.do',
+  'ebenefits/about/feature?feature=hearing-aid-batteries-and-prosthetic-socks':
+    'ebenefits/OrderMedicalEquip',
+  'ebenefits/about/feature?feature=sah-grant': 'ebenefits/SAH',
+  'ebenefits/about/feature?feature=vgli-policy-management':
+    'isam/sps/saml20idp/saml20/logininitial?PartnerId=https://fedsso-qa.prudential.com/cu&Target=https://giosgli-stage.prudential.com/osgli/Controller/eBenefitsUser',
+};
```

Comment:
Ah man, things like this are always very exciting to me...That feature query parameter would be awesome as a list of static constants, that could be passed as a prop into a container that would automatically translate the link to the SSO version -

```js
// ebenefits/features.js
export defaults const FEATURES = {
    DISABILITY_COMPENSATION: 'disability-compensation',
    VOCA_REHAB_AND_EMPLOYMENTS: 'vocational-rehabilitation-and-employment',
    DIRECT_DEPOSIT: 'direct-deposit-and-contact-information',
   //  .....
};

// ebenefits/sso-feature-map.js
export default const SSO_GATEWAY = new Map();

SSO_GATEWAY.set(
   FEATURES.DISABILITY_COMPENSATION, 
   'ebenefits/vdc?target=%2Fwssweb%2FVDC526%2Fcompensation.do'
);

SSO_GATEWAY.set(
    FEATURES.DIRECT_DEPOSIT,
    'ebenefits/manage/contact',
);
```

Then an app linking to eBenefits would only need to know the name of the feature using that static list and wouldn't need to worry about any Redux work in that application.

```jsx
import eBenefitsLink, { eBenefitsFeatures } from 'platform/ebenefits/containers/ebenefitsLink';

<eBenefitsLink feature={eBenefitsFeatures.DIRECT_DEPOSIT}>Go to direct deposit</eBenefitsLink>
```

Reply:
@ncksllvn I'll give it a shot.  I think the `feature` property would only apply to about half the links, so I'll just start with an `href` property.  After that if it looks like it can be further simplified we can iterate on it.

Reply:
@ncksllvn could you take another look at this PR?  I rewrote it to make use of a `ebenefitsLink` container.

Reply:
It looks great, fantastic work!